frontend/libs/studio-content-library/src/contexts/RouterContext.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { createContext, useContext } from 'react';
+import { useNavigation } from '../hooks/useNavigation';
+
+export type RouterContextProps = {
+  currentPage: string;
+  navigate: (page: string) => void;
+};
+
+export const RouterContext = createContext<Partial<RouterContextProps>>(undefined);
+
+export type RouterContextProviderProps = {
+  children: React.ReactNode;
+};
+
+export const RouterContextProvider = ({ children }: Partial<RouterContextProviderProps>) => {
+  const { navigate, currentPage } = useNavigation();
+
+  return (
+    <RouterContext.Provider
+      value={{
+        currentPage,
+        navigate,
+      }}
+    >
+      {children}
+    </RouterContext.Provider>
+  );
+};
+
+export const useRouterContext = (): Partial<RouterContextProps> => {
```

Comment:
Hvorfor `Partial`?

Reply:
Nice catch, det henger igjen her fra da jeg drev å testet konseptet og ikke ville ha midlertidig type feil på ting bare for å verifisere. Takk! 😊